compiler/AST/type.cpp

```diff
@@ -949,9 +947,8 @@ bool isLegalParamType(Type* t) {
 }
 
 int get_width(Type *t) {
-  if (t == dtBools[BOOL_SIZE_1] ||
-      t == dtBools[BOOL_SIZE_SYS]) {
-    return 1;
+  if (t == dtBools[BOOL_SIZE_SYS]) {
+    return 0;
```

Comment:
Actually I think we no longer need to lie here. (Of course this is a different issue.. but it might be worth addressing soon). Conceivably, the generated SysCTypes could include a declaration for c_bool and set it to `bool(8)` (or whatever it actually is). But, the SysCTypes logic wouldn't work for bool since there isn't a BOOL_MAX (as far as I know). But, AFAIK, with the LLVM backend, we can just set the bool size to 8 since we know that will be the case with clang.

Do you know of a current system where `sizeof(bool)` is not 8? If not, perhaps we should simply assert it is 8 in program startup and compile with that assumption?

Reply:
This value isn't used for anything other than a sentinel so that the code in `typeSpecifier.cpp` can map back to system-sized bool.  If we make this return 8, then we have a many to one mapping on the other side of things.